### PR TITLE
TXMNT-366 Migrated to play 2.5

### PR DESCRIPTION
**Pull request by franekrichardson**

### CODE REVIEW

**Comment by agilestevesmith** [comment]
Thanks @franekrichardson, @arminio will take this

File: src/main/scala/reactivemongo/json/collection/jsoncollection.scala [hunk 1]
```diff
@@ -186,13 +164,7 @@ object JSONBatchCommands
...
     def reads(js: JsValue): JsResult[Upserted] = for {
       ix <- (js \ "index").validate[Int]
-      id <- (js \ "_id") match {
-        case _: JsUndefined =>
-          JsError(__ \ "_id", "error.objectId")
-
-        case js =>
-          BSONFormats.toBSON(js)
-      }
+      id <- (js \ "_id").validate[JsValue].flatMap(BSONFormats.toBSON)
```
Comment (arminio) [inline src/main/scala/reactivemongo/json/collection/jsoncollection.scala]: This used to return JsError in the default case, we are not doing that anymore. is that intentional?

Comment (franekrichardson) [inline src/main/scala/reactivemongo/json/collection/jsoncollection.scala]: So my process for migrating quite a lot of this was to use the mainline reactivemongo repository as a guide - as it is compatible with play 2.4/2.5.
In this case I suspect in this case they assume the _id filed is always present so have removed the error case.

Comment (franekrichardson) [inline src/main/scala/reactivemongo/json/collection/jsoncollection.scala]: So my process for migrating quite a lot of this was to use the mainline reactivemongo repository as a guide - as it is compatible with play 2.4/2.5.
In this case I suspect in this case they assume the _id filed is always present so have removed the error case.

**Pull request merged**